### PR TITLE
[EN] Immutant migration

### PR DESCRIPTION
**Pull request by ericnormand**
https://www.pivotaltracker.com/story/show/90560506

```
> docker-compose build
> docker-compose up
```

`docker-compose up` requires the following ENV_VARS:

`WILDFLY_ADMIN_USER` and `WILDFLY_ADMIN_PASSWORD`, with values from
https://github.com/democracyworks/dockerfiles/blob/master/wildfly/Dockerfile#L4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
-FIXME
+```
+> docker-compose build
+> docker-compose up
+```
+
+`docker-compose up` requires the following ENV_VARS:
+
+`WILDFLY_ADMIN_USER` and `WILDFLY_ADMIN_PASSWORD`, with values from
+https://github.com/democracyworks/dockerfiles/blob/master/wildfly/Dockerfile#L4
```
Comment (cap10morgan) [inline README.md]: We have moved the wildfly image into [its own repo](https://github.com/democracyworks/wildfly-coreos) and the admin creds are now set via the `ADMIN_USERNAME` and `ADMIN_PASSWORD` env vars when running that container. So matching values for those env vars and these could just be baked into the docker-compose.yml here. They really just need to agree in each environment.

File: project.clj [hunk 2]
```diff
@@ -4,12 +4,11 @@
...
-                 [http-kit "2.1.16"]
+                 [org.immutant/web "2.0.0-beta2"]
                  [compojure "1.1.6"]
-                 [javax.servlet/servlet-api "2.5"]
                  [ring/ring-json "0.3.1"]
                  [riemann-clojure-client "0.2.10"]]
   :main sendgrid-monitor.core
   :profiles {:uberjar {:aot [sendgrid-monitor.core]}}
-  :plugins [[lein-immutant "1.2.0"]]
+  :plugins [[lein-immutant "2.0.0-beta1"]]
```
Comment (cap10morgan) [inline project.clj]: Looks like [2.0.0 has been released](https://clojars.org/lein-immutant).

File: README.md (hunk 1 above)
Comment (ericnormand) [inline README.md]: Hey @cap10morgan , I for sure don't know how to do that in docker-compose. Is there anywhere you can show me where this is done?

Comment (cap10morgan) [inline README.md]: Like this:

```yaml
app:
  build: .
  environment:
    WILDFLY_ADMIN_USER: admin
    WILDFLY_ADMIN_PASSWORD: admin
  links:
    - wildfly:wildfly
wildfly:
  image: quay.io/democracyworks/wildfly:8.2.0.Final
  environment:
    ADMIN_USERNAME: admin
    ADMIN_PASSWORD: admin
```

Comment (ericnormand) [inline README.md]: Oh, I see what you mean.

Comment (tie-rack) [inline README.md]: I believe there's a better way than copying and pasting values. Aren't env vars in one docker container available in linked containers with a prefix?

**Comment by ericnormand** [comment]
ping @cap10morgan 

Fixed your two comments.

File: README.md (hunk 1 above)
Comment (cap10morgan) [inline README.md]: @tie-rack I don't know but if they are that would be better.

File: project.clj [hunk 3]
```diff
@@ -4,12 +4,11 @@
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
   :dependencies [[org.clojure/clojure "1.6.0"]
-                 [http-kit "2.1.16"]
+                 [org.immutant/web "2.0.0-beta2"]
```
Comment (cap10morgan) [inline project.clj]: Might as well bump this to 2.0.0-beta3 since it's out.

File: README.md (hunk 1 above)
Comment (ericnormand) [inline README.md]: @tie-rack I agree. But I don't see anything like that in the docker-compose docs: https://docs.docker.com/compose/yml/#environment

Comment (tie-rack) [inline README.md]: It comes from docker links (which docker-compose is doing):

> Docker also exposes each Docker originated environment variable from the source container as an environment variable in the target. For each variable Docker creates an `<alias>_ENV_<name>` variable in the target container. The variable's value is set to the value Docker used when it started the source container.

From the [Environment Variables section of the Docker documentation on links](docs.docker.com/userguide/dockerlinks/#environment-variables).

So, those wildfly env vars would be available in sendgrid-monitor as `WILDFLY_ENV_ADMIN_USERNAME` and `WILDFLY_ENV_ADMIN_PASSWORD`.

Comment (ericnormand) [inline README.md]: Ok, so this would require a change in didor to look at those values instead. Which may be the best, general solution.

Comment (tie-rack) [inline README.md]: Yeah. I think that makes sense.

Comment (ericnormand) [inline README.md]: Ok, I made a [PR on didor](https://github.com/democracyworks/didor/pull/4) for that change.

**Comment by ericnormand** [comment]
This PR is now dependent on [this PR on didor](https://github.com/democracyworks/didor/pull/4)

**Comment by cap10morgan** [comment]
And now Immutant 2.0.0 is out. Please to update?

**Comment by ericnormand** [comment]
Upgraded to Immutant 2

Ping @cap10morgan

**Comment by cap10morgan** [comment]
This LGTM. Merging to master will render this inoperable in the old docker cluster, but we're re-evaluating our monitoring and sendgrid usage anyway. So here goes!

**Pull request merged**